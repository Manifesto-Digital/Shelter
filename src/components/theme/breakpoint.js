import { css } from 'styled-components';

const sizes = {
  desktopWide: 1170,
  desktop: 992,
  tabletWide: 741,
  tablet: 641,
  mobileLand: 512,
  mobile: 359,
};

// iterate through the sizes and create a media template
export const breakpoint = Object.keys(sizes).reduce((accumulator, label) => {
  // use em in breakpoints to work properly cross-browser and support users
  // changing their browsers font-size: https://zellwk.com/blog/media-query-units/
  const emSize = sizes[label] / 16;
  accumulator[label] = (...args) => css`
    @media (min-width: ${emSize}em) {
      ${css(...args)};
    }
  `;
  return accumulator;
}, {});
