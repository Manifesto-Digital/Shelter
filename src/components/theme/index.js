const theme = {}

theme.palette = {
  primary: '#EA232D',
  secondary: 'DBC8B6',
  donate: '12b9bf',
  link: '#4d63b5',
  danger: '#ff0000',
  success: '#72dcd4',
  white: '#fff',
  black: '#212122',
  greyLight: '#e5e5e5',
  greyMed: '#d9d9d9',
}

theme.fonts = {
  primary: 'Helvetica Neue, Helvetica, Roboto, sans-serif',
  header: 'Consolas, Liberation Mono, Menlo, Courier, monospace',
  quote: 'Georgia, serif',
}

theme.sizes = {
  maxWidth: '1100px',
}

theme.spacing = {
  padding: '20px',
  largePadding: '40px',
}
export default theme
