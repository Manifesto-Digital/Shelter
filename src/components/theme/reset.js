export const reset = `
*,
:: after,
:: before {
<<<<<<< HEAD
    box - sizing: border - box;
}
html {
    font - family: sans - serif;
    line - height: 1.15;
    -webkit - text - size - adjust: 100 %;
    -ms - text - size - adjust: 100 %;
    -ms - overflow - style: scrollbar;
    -webkit - tap - highlight - color: transparent;
}
@-ms - viewport {
    width: device - width;
=======
    box-sizing: border-box;
}
html {
    font-family: sans-serif;
    line-height: 1.15;
    -webkit-text-size-adjust: 100 %;
    -ms-text-size-adjust: 100 %;
    -ms-overflow-style: scrollbar;
    -webkit-tap-highlight-color: transparent;
}
@-ms-viewport {
    width: device-width;
>>>>>>> cd4ba4249a6699a6ba039d07ff006ab8f1bec027
}
article,
    aside,
    figcaption,
    figure,
    footer,
    header,
    hgroup,
    main,
    nav,
    section {
    display: block;
}
body {
    margin: 0;
<<<<<<< HEAD
    font - size: 1rem;
    font - weight: 400;
    line - height: 1.5;
    color: #212529;
    text - align: left;
    background - color: #fff;
=======
    font-size: 1rem;
    font-weight: 400;
    line-height: 1.5;
    color: #212529;
    text-align: left;
    background-color: #fff;
>>>>>>> cd4ba4249a6699a6ba039d07ff006ab8f1bec027
}
[tabindex = '-1']: focus {
    outline: 0!important;
}
hr {
<<<<<<< HEAD
    box - sizing: content - box;
=======
    box-sizing: content-box;
>>>>>>> cd4ba4249a6699a6ba039d07ff006ab8f1bec027
    height: 0;
    overflow: visible;
}
h1,
    h2,
    h3,
    h4,
    h5,
    h6 {
<<<<<<< HEAD
    margin - top: 0;
    margin - bottom: 0.5rem;
}
p {
    margin - top: 0;
    margin - bottom: 1rem;
}
abbr[data - original - title],
    abbr[title] {
    text - decoration: underline;
    -webkit - text - decoration: underline dotted;
    text - decoration: underline dotted;
    cursor: help;
    border - bottom: 0;
}
address {
    margin - bottom: 1rem;
    font - style: normal;
    line - height: inherit;
=======
    margin-top: 0;
    margin-bottom: 1rem;
}
p {
    margin-top: 0;
    margin-bottom: 1rem;
}
abbr[data-original-title],
    abbr[title] {
    text-decoration: underline;
    -webkit-text-decoration: underline dotted;
    text-decoration: underline dotted;
    cursor: help;
    border-bottom: 0;
}
address {
    margin-bottom: 1rem;
    font-style: normal;
    line-height: inherit;
>>>>>>> cd4ba4249a6699a6ba039d07ff006ab8f1bec027
}
dl,
    ol,
    ul {
<<<<<<< HEAD
    margin - top: 0;
    margin - bottom: 1rem;
=======
    margin-top: 0;
    margin-bottom: 1rem;
>>>>>>> cd4ba4249a6699a6ba039d07ff006ab8f1bec027
}
ol ol,
    ol ul,
        ul ol,
            ul ul {
<<<<<<< HEAD
    margin - bottom: 0;
}
dt {
    font - weight: 700;
}
dd {
    margin - bottom: 0.5rem;
    margin - left: 0;
=======
    margin-bottom: 0;
}
dt {
    font-weight: 700;
}
dd {
    margin-bottom: 0.5rem;
    margin-left: 0;
>>>>>>> cd4ba4249a6699a6ba039d07ff006ab8f1bec027
}
blockquote {
    margin: 0 0 1rem;
}
dfn {
<<<<<<< HEAD
    font - style: italic;
}
b,
    strong {
    font - weight: bolder;
}
small {
    font - size: 80 %;
=======
    font-style: italic;
}
b,
    strong {
    font-weight: bolder;
}
small {
    font-size: 80 %;
>>>>>>> cd4ba4249a6699a6ba039d07ff006ab8f1bec027
}
sub,
    sup {
    position: relative;
<<<<<<< HEAD
    font - size: 75 %;
    line - height: 0;
    vertical - align: baseline;
=======
    font-size: 75 %;
    line-height: 0;
    vertical-align: baseline;
>>>>>>> cd4ba4249a6699a6ba039d07ff006ab8f1bec027
}
sub {
    bottom: -0.25em;
}
sup {
    top: -0.5em;
}
a {
<<<<<<< HEAD
    text - decoration: none;
    background - color: transparent;
    -webkit - text - decoration - skip: objects;
}
a: hover {
    text - decoration: underline;
}
a: not([href]): not([tabindex]) {
    color: inherit;
    text - decoration: none;
=======
    text-decoration: none;
    background-color: transparent;
    -webkit-text-decoration-skip: objects;
}
a: hover {
    text-decoration: underline;
}
a: not([href]): not([tabindex]) {
    color: inherit;
    text-decoration: none;
>>>>>>> cd4ba4249a6699a6ba039d07ff006ab8f1bec027
}
a: not([href]): not([tabindex]): focus,
    a: not([href]): not([tabindex]): hover {
    color: inherit;
<<<<<<< HEAD
    text - decoration: none;
=======
    text-decoration: none;
>>>>>>> cd4ba4249a6699a6ba039d07ff006ab8f1bec027
}
a: not([href]): not([tabindex]): focus {
    outline: 0;
}
code,
    kbd,
    pre,
    samp {
<<<<<<< HEAD
    font - family: SFMono - Regular, Menlo, Monaco, Consolas, 'Liberation Mono', 'Courier New', monospace;
    font - size: 1em;
}
pre {
    margin - top: 0;
    margin - bottom: 1rem;
    overflow: auto;
    -ms - overflow - style: scrollbar;
=======
    font-family: SFMono-Regular, Menlo, Monaco, Consolas, 'Liberation Mono', 'Courier New', monospace;
    font-size: 1em;
}
pre {
    margin-top: 0;
    margin-bottom: 1rem;
    overflow: auto;
    -ms-overflow-style: scrollbar;
>>>>>>> cd4ba4249a6699a6ba039d07ff006ab8f1bec027
}
figure {
    margin: 0 0 1rem;
}
img {
<<<<<<< HEAD
    vertical - align: middle;
    border - style: none;
}
svg {
    overflow: hidden;
    vertical - align: middle;
}
table {
    border - collapse: collapse;
}
caption {
    padding - top: 0.75rem;
    padding - bottom: 0.75rem;
    color: #6c757d;
    text - align: left;
    caption - side: bottom;
}
th {
    text - align: inherit;
}
label {
    display: inline - block;
    margin - bottom: 0.5rem;
}
button {
    border - radius: 0;
=======
    vertical-align: middle;
    border-style: none;
}
svg {
    overflow: hidden;
    vertical-align: middle;
}
table {
    border-collapse: collapse;
}
caption {
    padding-top: 0.75rem;
    padding-bottom: 0.75rem;
    color: #6c757d;
    text-align: left;
    caption-side: bottom;
}
th {
    text-align: inherit;
}
label {
    display: inline-block;
    margin-bottom: 0.5rem;
}
button {
    border-radius: 0;
>>>>>>> cd4ba4249a6699a6ba039d07ff006ab8f1bec027
}

button,
    input,
    optgroup,
    select,
    textarea {
    margin: 0;
<<<<<<< HEAD
    font - family: inherit;
    font - size: inherit;
    line - height: inherit;
=======
    font-family: inherit;
    font-size: inherit;
    line-height: inherit;
>>>>>>> cd4ba4249a6699a6ba039d07ff006ab8f1bec027
}
button,
    input {
    overflow: visible;
}
button,
    select {
<<<<<<< HEAD
    text - transform: none;
=======
    text-transform: none;
>>>>>>> cd4ba4249a6699a6ba039d07ff006ab8f1bec027
}
[type = 'reset'],
    [type = 'submit'],
    button,
    html[type = 'button'] {
<<<<<<< HEAD
    -webkit - appearance: button;
}
[type = 'button']:: -moz - focus - inner,
    [type = 'reset']:: -moz - focus - inner,
        [type = 'submit']:: -moz - focus - inner,
            button:: -moz - focus - inner {
    padding: 0;
    border - style: none;
}
input[type = 'checkbox'],
    input[type = 'radio'] {
    box - sizing: border - box;
=======
    -webkit-appearance: button;
}
[type = 'button']:: -moz-focus-inner,
    [type = 'reset']:: -moz-focus-inner,
        [type = 'submit']:: -moz-focus-inner,
            button:: -moz-focus-inner {
    padding: 0;
    border-style: none;
}
input[type = 'checkbox'],
    input[type = 'radio'] {
    box-sizing: border-box;
>>>>>>> cd4ba4249a6699a6ba039d07ff006ab8f1bec027
    padding: 0;
}
input[type = 'date'],
    input[type = 'datetime-local'],
    input[type = 'month'],
    input[type = 'time'] {
<<<<<<< HEAD
    -webkit - appearance: listbox;
=======
    -webkit-appearance: listbox;
>>>>>>> cd4ba4249a6699a6ba039d07ff006ab8f1bec027
}
textarea {
    overflow: auto;
    resize: vertical;
}
fieldset {
<<<<<<< HEAD
    min - width: 0;
=======
    min-width: 0;
>>>>>>> cd4ba4249a6699a6ba039d07ff006ab8f1bec027
    padding: 0;
    margin: 0;
    border: 0;
}
legend {
    display: block;
    width: 100 %;
<<<<<<< HEAD
    max - width: 100 %;
    padding: 0;
    margin - bottom: 0.5rem;
    font - size: 1.5rem;
    line - height: inherit;
    color: inherit;
    white - space: normal;
}
progress {
    vertical - align: baseline;
}
[type = 'number']:: -webkit - inner - spin - button,
    [type = 'number']:: -webkit - outer - spin - button {
    height: auto;
}
[type = 'search'] {
    outline - offset: -2px;
    -webkit - appearance: none;
}
[type = 'search']:: -webkit - search - cancel - button,
    [type = 'search']:: -webkit - search - decoration {
    -webkit - appearance: none;
}
:: -webkit - file - upload - button {
    font: inherit;
    -webkit - appearance: button;
}
output {
    display: inline - block;
}
summary {
    display: list - item;
=======
    max-width: 100 %;
    padding: 0;
    margin-bottom: 0.5rem;
    font-size: 1.5rem;
    line-height: inherit;
    color: inherit;
    white-space: normal;
}
progress {
    vertical-align: baseline;
}
[type = 'number']:: -webkit-inner-spin-button,
    [type = 'number']:: -webkit-outer-spin-button {
    height: auto;
}
[type = 'search'] {
    outline-offset: -2px;
    -webkit-appearance: none;
}
[type = 'search']:: -webkit-search-cancel-button,
    [type = 'search']:: -webkit-search-decoration {
    -webkit-appearance: none;
}
:: -webkit-file-upload-button {
    font: inherit;
    -webkit-appearance: button;
}
output {
    display: inline-block;
}
summary {
    display: list-item;
>>>>>>> cd4ba4249a6699a6ba039d07ff006ab8f1bec027
    cursor: pointer;
}
template {
    display: none;
}
[hidden] {
    display: none!important;
}
/*# sourceMappingURL=bootstrap-reboot.min.css.map */
`
