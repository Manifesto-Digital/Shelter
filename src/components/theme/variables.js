const theme = {}

theme.palette = {
  primary: '#EA232D',
  secondary: '#DBC8B6',
  donate: '#12b9bf',
  link: '#4d63b5',
  danger: '#ff0000',
  success: '#72dcd4',
  white: '#fff',
  black: '#212122',
  greyLight: '#e5e5e5',
  greyMed: '#d9d9d9',
}

theme.fonts = {
  primary: 'Helvetica Neue, Helvetica, Roboto, sans-serif',
  header: 'Helvetica Neue, Helvetica, arial,sans-serif',
  quote: 'Georgia, serif',
}

theme.sizes = {
  maxWidth: '1100px',
}

theme.headers = {
  h1: '3em',
  h2: '2.25em',
  h3: '1.75em',
  h4: '1.125em',
}

theme.spacing = {
  padding: '20px',
  largePadding: '40px',
}
export default theme
