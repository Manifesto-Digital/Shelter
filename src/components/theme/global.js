import { createGlobalStyle } from 'styled-components';
import { reset } from './reset';
import { breakpoint } from './breakpoint';
import { linkStyles } from '../styled/links';

export const GlobalStyle = createGlobalStyle`
    ${reset}
    html {
        height: 100%;
        font-size:16px;
        scroll-behavior: smooth !important;
        ${breakpoint.tablet`
            font-size:  115%;
        `};
    }

    body {
        min-height: 100%;
        height: 100%;
        margin: 0;
        padding: 0;
        font-family: ${props => props.theme.fonts.primary};
        color: ${props => props.theme.palette.black};
    }

    h1,
    h2,
    h3,
    h4,
    h5,
    h6 {
        font-family: ${props => props.theme.fonts.header};
    }

    h1, h2, h3 {
        line-height: 1.25;
    }



    h1 {
        font-size:  ${props => props.theme.headers.h2};
        ${breakpoint.tablet`
            font-size:  ${props => props.theme.headers.h1};
            line-height: 1.15;
        `};
    }

    h2 {
        font-size:  ${props => props.theme.headers.h3};
        ${breakpoint.tablet`
            font-size:  ${props => props.theme.headers.h2};
        `};
    }
    h3 {
        font-size:  ${props => props.theme.headers.h4};
        ${breakpoint.tablet`
            font-size:  ${props => props.theme.headers.h3};
        `};
    }

    img {
        max-width: 100%;
    }

    a {
        ${linkStyles};
    }

    li {
        margin-bottom: .5rem;
    }

    input{
        border: 1px solid ${props => props.theme.palette.grey15};
    }

    button{
        border: none;
    }
`;
