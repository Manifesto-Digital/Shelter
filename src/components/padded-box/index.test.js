import React from 'react';
import {
  snapshotComponent,
  renderWithTheme,
} from '../../../__tests__/helpers/index';
import theme from '../theme/variables';

import PaddedBox from '.';

it('renders correctly with defaults', () => {
  snapshotComponent(<PaddedBox />);
});

it('changes based on supplied props', () => {
  const wrapper = renderWithTheme(
    <PaddedBox bg="grey45" padding="large" removeMarginBottom />
  );
  expect(wrapper.toJSON()).toHaveStyleRule('padding', theme.spacing.large);
  expect(wrapper.toJSON()).toHaveStyleRule(
    'background-color',
    theme.palette.grey45
  );
  expect(wrapper.toJSON()).toHaveStyleRule('margin-bottom', '0');

  const wrapper2 = renderWithTheme(
    <PaddedBox bg="sanMarinoBlue" padding="small" />
  );

  expect(wrapper2.toJSON()).toHaveStyleRule('padding', theme.spacing.small);
  expect(wrapper2.toJSON()).toHaveStyleRule(
    'background-color',
    theme.palette.sanMarinoBlue
  );
});

it('it renders with shadow when supplied', () => {
  const wrapper = renderWithTheme(<PaddedBox shadow />);
  expect(wrapper.toJSON()).toHaveStyleRule('box-shadow', theme.boxshadow.small);
  expect(wrapper.toJSON()).toHaveStyleRule(
    'border-radius',
    theme.borderradius.small
  );
});
