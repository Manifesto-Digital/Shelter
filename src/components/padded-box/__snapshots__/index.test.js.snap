// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with defaults 1`] = `
.c0 {
  padding: 20px;
  background-color: #f7f7f7;
  box-shadow: 0;
  border-radius: 0;
}

<div
  className="c0"
/>
`;
